core/shared/src/main/scala/zio/config/syntax/ConfigSyntax.scala

```diff
@@ -150,14 +150,15 @@ trait ConfigSyntax {
       def loop[B](config: Config[B]): Config[B] =
         config match {
           case Described(config, description) => Described(loop(config), description)
-          case config: FallbackWith[B] => FallbackWith(loop(config.first), loop(config.second), config.f)
-          case config: Fallback[B] => Fallback(loop(config.first), loop(config.second))
-          case Sequence(config) => Sequence(loop(config))
-          case Nested(name, config) => Nested(f(name), loop(config))
+          case config: FallbackWith[B]        => FallbackWith(loop(config.first), loop(config.second), config.f)
+          case config: Fallback[B]            => Fallback(loop(config.first), loop(config.second))
+          case Sequence(config)               => Sequence(loop(config))
+          case Switch(config, map)            => Switch(config, map.map { case (k, v) => k -> loop(v) })
```

Comment:
BTW, I wonder if `Config#mapKey` extension method is still necessary, because now I found it more idiomatic to adjust the keys using `ConfigProvider`. (e.g. `ConfigProvider#kebabCase`)

Reply:
I think we will keep it for now.